src/custom/hooks/useAllTransactionsAndOrders.ts

```diff
@@ -0,0 +1,59 @@
+import { useMemo } from 'react'
+import { useAllTransactions } from 'state/transactions/hooks'
+import { useOrders } from 'state/orders/hooks'
+import { useActiveWeb3React } from 'hooks'
+import { Order, OrderStatus } from 'state/orders/actions'
+import { TransactionDetails } from '@src/state/transactions/reducer'
+
+type TransactionAndOrder =
+  | (Order & { addedTime: number })
+  | (TransactionDetails & {
+      id: string
+      status: 'pending' | 'fulfilled'
+    })
+
+/**
+ * Returns whether a transaction/order happened in the last day (86400 seconds * 1000 milliseconds / second)
+ * @param time to check for recency
+ */
+export function isRecent(time: number): boolean {
+  return new Date().getTime() - time < 86_400_000
+}
+
+function isPending(data: TransactionAndOrder) {
+  return data.status === OrderStatus.PENDING
+}
+
+function isFulfilled(data: TransactionAndOrder) {
+  return data.status === OrderStatus.FULFILLED
+}
+
+export default function useAllTransactionsAndOrders() {
+  const { chainId } = useActiveWeb3React()
+  const allTransactions = useAllTransactions()
+  const allNonEmptyOrders = useOrders({ chainId })
+
+  return useMemo(() => {
+    const adjustedTransactions = Object.values(allTransactions)
+      .map(tx => ({ ...tx, id: tx.hash, status: tx.receipt ? 'fulfilled' : 'pending' }))
+      .filter(tx => isRecent(tx.addedTime))
+
+    const adjustedOrders = allNonEmptyOrders
+      .map(order => ({ ...order, addedTime: Date.parse(order.creationTime) }))
+      .filter(order => isRecent(order.addedTime))
+
+    const sortedTransactionsAndOrders = (adjustedTransactions as TransactionAndOrder[])
+      .concat(adjustedOrders)
+      .sort((a, b) => {
+        return b.addedTime - a.addedTime
+      })
+
+    const pendingTransactions = sortedTransactionsAndOrders.filter(isPending).map(data => data.id)
+    const confirmedTransactions = sortedTransactionsAndOrders.filter(isFulfilled).map(data => data.id)
```

Comment:
In general i feel this hook has a lot of logic, right. Is his responsibility for example remove old orders, or there should be a maintenance tasks updating the state?

Reply:
looking again i agree, maybe keep the name, but don't filter old orders? just combine the two. But the point of this hook is to get all orders AND transactions together and filter for just the recent ones. so 2 options i see:
1. keep hook name and return all orders + transactions as two arrays: 
  a. `pendingTransactionsAndOrders` or `pendingActivity` (or some other name)
  b. `fulfilledTransactionsAndOrders` or `fulfilledActivity` (or some other name)
2. change name of hook to `useAllRecentTransactionsAndOrders` or `useAllRecentActivity` (or some other name)
  a. return the same logic as is